pandas/tests/internals/test_internals.py

```diff
@@ -692,18 +692,20 @@ def test_get_bool_data(self):
             bools.iget(bools.items.get_loc("bool")).internal_values(),
         )
 
+        # GH#33457 setting a new array on bools does _not_ alter the original
+        #  array in-place, so mgr is unchanged
         bools.iset(0, np.array([True, False, True]))
         tm.assert_numpy_array_equal(
             mgr.iget(mgr.items.get_loc("bool")).internal_values(),
-            np.array([True, False, True]),
+            np.array([True, True, True]),
         )
 
-        # Check sharing
+        # Check non-sharing
         bools2 = mgr.get_bool_data(copy=True)
-        bools2.iset(0, np.array([False, True, False]))
+        bools2.blocks[0].values[:] = [[False, True, False]]
```

Comment:
can you split the non-sharing to a new test

Reply:
updated  + green